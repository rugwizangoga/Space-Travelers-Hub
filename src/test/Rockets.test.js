import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/pages/Rockets';
import store from '../redux/configureStore';

jest.mock('axios', () => ({
  post: jest.fn(),
  get: jest.fn(),
}));

describe('Rockets component', () => {
  it('should render correctly', () => {
    const rocket = [
      {
        id: 'rocket1',
        name: 'Rocket 1',
        description: 'Description 1',
        flickr_images: ['image1'],
        reserved: false,
      },
    ];

    const RocketsList = TestRenderer.create(
      <Provider store={store}>
        <Rockets key={rocket.id} RocketsList={rocket} />
      </Provider>,
    ).toJSON();
    expect(RocketsList).toMatchSnapshot();
  });
});
