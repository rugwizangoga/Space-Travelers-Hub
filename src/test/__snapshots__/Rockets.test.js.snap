// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component should render correctly 1`] = `
<div
  className="rocket-container"
/>
`;
